webapp-django/crashstats/api/tests/test_views.py

```diff
@@ -205,180 +204,145 @@ def test_hit_or_not_hit_ratelimit(self):
             assert response.status_code == 429
 
     def test_ProcessedCrash(self):
-        url = reverse("api:model_wrapper", args=("ProcessedCrash",))
-        response = self.client.get(url)
-        assert response.status_code == 400
-        assert response["Content-Type"] == "application/json"
-        dump = json.loads(response.content)
-        assert dump["errors"]["crash_id"]
+        public_data = {
+            "addons_checked": None,
+            "address": "0x8",
+            "build": "20120609030536",
+            "client_crash_date": "2012-06-11T06:08:45",
+            "completed_datetime": "2012-06-11T06:08:57",
+            "cpu_arch": "amd64",
+            "date_processed": "2012-06-11T06:08:44",
+            "last_crash": 371342,
+            "os_name": "Mac OS X",
+            "os_version": "10.6.8 10K549",
+            "process_type": None,
+            "product": "WaterWolf",
+            "reason": "EXC_BAD_ACCESS / KERN_INVALID_ADDRESS",
+            "release_channel": "nightly",
+            "release_channel": "nightly",
+            "signature": "FakeSignature1",
+            "success": True,
+            "uptime": 14693,
+            "uuid": "11cb72f5-eb28-41e1-a8e4-849982120611",
+            "version": "5.0a1",
+        }
+
+        protected_data = {
+            "url": "https://example.com",
+            "user_comments": None,
```

Comment:
This structure is _so_ much clearer about what's going on and what's correct and not correct.